doc.go

```diff
@@ -63,58 +64,61 @@ Start skipper and make an HTTP request through skipper:
+route. The response from the backend goes once again through all the
+filters, but this time the order is reversed, then it finally gets
+mapped as the response of the original incoming request.
 
 Besides the default proxying mechanism, it is possible to define routes
 without a real network backend endpoint. This is called a 'shunt'
 backend, in which case one of the filters needs to handle the request
-(e.g. the 'static' filter).
+(e.g. the 'static' filter). Actually, filters themselves can instruct
+the request flow to shunt.
 
-For details, see the documentation of the proxy subdirectory.
+For further details, see the 'proxy' and 'filters' package
+documentation.
 
 
 Matching Requests
 
-Finding the route for a request happens by matching the request
-attributes against the conditions in the route definitions. Route
-definitions may have the following conditions: method, path (optionally
-with wildcards), path regular expressions, host regular expressions,
-headers and header regular expressions. Besides, it is possible to
-define custom predicates, implemented as extensions.
+Finding a request's route happens by matching the request attributes to
+the conditions in the route's definitions. Such definitions may have the
+following conditions: method, path (optionally with wildcards), path
+regular expressions, host regular expressions, headers and header
+regular expressions. There is also a way to use custom predicates as
+conditions to be matched.
 
 The relation between the conditions in a route definition is 'and',
-meaning that a request must fulfil each condition to match a route.
+meaning, that a request must fulfill each condition to match a route.
 
-For details, see the documentation of the routing subdirectory.
+For further details, see the 'routing' package documentation.
 
 
 Filters - Augmenting Requests
 
-Filters are applied in order of definition to the request and in
-reverse order to the response. They are used to modify request and
-response attributes like headers, or execute background tasks, like
-logging. Some filters may handle the requests without proxying them to
-service backends. Filters, depending on their implementation, may
-accept/require parameters, that are set specific to the route.
+Filters are applied in order of definition to the request and in reverse
+order to the response. They are used to modify request and response
+attributes, such as headers, or execute background tasks, like logging.
+Some filters may handle the requests without proxying them to service
+backends. Filters, depending on their implementation, may accept/require
+parameters, that are set specific to the route.
 
-For details, see the documentation of the filters subdirectory.
+For details, see the 'filters' package documentation.
 
 
 Service Backends
 
-Each route has a backend, one of two kinds: network service or shunt.
+Each route has one of the following backends: network service or shunt.
```

Comment:
network service, specifically, HTTP, right? Would it be better to state so?
"HTTP endpoint and shunt" ?